homeassistant/components/sonos/__init__.py

```diff
@@ -122,10 +124,12 @@ def _discovered_player(soco: SoCo) -> None:
                 data = hass.data[DATA_SONOS]
 
                 if soco.uid not in data.discovered:
-                    _LOGGER.debug("Adding new speaker")
                     speaker_info = soco.get_speaker_info(True)
+                    _LOGGER.debug("Adding new speaker: %s", speaker_info)
                     speaker = SonosSpeaker(hass, soco, speaker_info)
                     data.discovered[soco.uid] = speaker
+                    data.favorites.setdefault(soco.household_id, SonosFavorites(soco))
```

Comment:
Can the household_id be different?  I guess you could have two systems on the same network??

Reply:
Yes, it's rare but can happen. For example, if you have separate S1 and S2 systems on the same network.